streams-tests/jvm/src/test/scala/zio/stream/SinkSpec.scala

```diff
@@ -393,6 +399,26 @@ class SinkSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRunt
     unsafeRun(sinkIteration(sink, Chunk.single(1)).either.map(_ must_=== Left("Ouch")))
   }
 
+  private def chunkedLeftover = {
+    val sink = ZSink.collectAllN[Int](2).chunked
+    val test = for {
+      init   <- sink.initial
+      step   <- sink.step(init, Chunk(1, 2, 3, 4, 5))
+      result <- sink.extract(step)
+    } yield result must_=== ((List(1, 2), Chunk(3, 4, 5)))
+    unsafeRun(test)
+  }
+
+  private def chunkedLeftoverAppend = {
+    val sink = ZSink.ignoreWhile[Int](_ < 0).chunked
```

Comment:
You know, I think the semantics of `ignoreWhile` should not be "return ignored elements as leftover", but rather "swallow elements that are ignored". What do you think?

Reply:
Technically, it returns the first non ignored element as leftover.

Reply:
Oh, sorry, I misread the test. All good then.

Reply:
Maybe it would be better as `dropUntil` with a flipped predicate, but it's more or less the same.